Makefile

```diff
@@ -172,102 +209,129 @@ generate-operator-yaml: kustomize manifests
-	kubectl create ns managed --kubeconfig=$(PWD)/kubeconfig_hub
+	kubectl create ns $(WATCH_NAMESPACE) --kubeconfig=$(HUB_CONFIG)
 	@echo creating namespace on managed
-	kubectl create ns managed --kubeconfig=$(PWD)/kubeconfig_managed
+	kubectl create ns $(WATCH_NAMESPACE) --kubeconfig=$(MANAGED_CONFIG)
 
+.PHONY: e2e-dependencies
 e2e-dependencies:
-	go get github.com/onsi/ginkgo/v2/ginkgo@$(GINKGO_VERSION)
-	go get github.com/onsi/gomega/...@$(GOMEGA_VERSION)
+	$(call go-get-tool,github.com/onsi/ginkgo/v2/ginkgo@$(shell awk '/github.com\/onsi\/ginkgo\/v2/ {print $$2}' go.mod))
+
+.PHONY: e2e-test
+e2e-test: e2e-dependencies
+	$(GINKGO) -v --fail-fast --slow-spec-threshold=10s $(E2E_TEST_ARGS) test/e2e
 
-e2e-test:
-	ginkgo -v --slowSpecThreshold=10 test/e2e
+.PHONY: e2e-test-coverage
+e2e-test-coverage: E2E_TEST_ARGS = --json-report=report_e2e.json --output-dir=.
+e2e-test-coverage: e2e-run-instrumented e2e-test e2e-stop-instrumented
 
+.PHONY: e2e-build-instrumented
+e2e-build-instrumented:
+	go test -covermode=atomic -coverpkg=$(shell cat go.mod | head -1 | cut -d ' ' -f 2)/... -c -tags e2e ./ -o build/_output/bin/$(IMG)-instrumented
+
+.PHONY: e2e-run-instrumented
+e2e-run-instrumented: e2e-build-instrumented
+	HUB_CONFIG=$(HUB_CONFIG) MANAGED_CONFIG=$(MANAGED_CONFIG) WATCH_NAMESPACE=$(WATCH_NAMESPACE) ./build/_output/bin/$(IMG)-instrumented -test.run "^TestRunMain$$" -test.coverprofile=coverage_e2e.out &>/dev/null &
+
+.PHONY: e2e-stop-instrumented
+e2e-stop-instrumented:
+	ps -ef | grep '$(IMG)' | grep -v grep | awk '{print $$2}' | xargs kill
+
+.PHONY: e2e-debug
 e2e-debug:
 	@echo gathering hub info
-	kubectl get all -n managed --kubeconfig=$(PWD)/kubeconfig_hub
-	kubectl get Policy.policy.open-cluster-management.io --all-namespaces --kubeconfig=$(PWD)/kubeconfig_hub
+	kubectl get all -n managed --kubeconfig=$(HUB_CONFIG)
+	kubectl get Policy.policy.open-cluster-management.io --all-namespaces --kubeconfig=$(HUB_CONFIG)
 	@echo gathering managed cluster info
-	kubectl get all -n $(KIND_NAMESPACE) --kubeconfig=$(PWD)/kubeconfig_managed
-	kubectl get all -n managed --kubeconfig=$(PWD)/kubeconfig_managed
-	kubectl get Policy.policy.open-cluster-management.io --all-namespaces --kubeconfig=$(PWD)/kubeconfig_managed
-	kubectl describe pods -n $(KIND_NAMESPACE) --kubeconfig=$(PWD)/kubeconfig_managed
-	kubectl logs $$(kubectl get pods -n $(KIND_NAMESPACE) -o name --kubeconfig=$(PWD)/kubeconfig_managed | grep $(IMG)) -n $(KIND_NAMESPACE) --kubeconfig=$(PWD)/kubeconfig_managed
+	kubectl get all -n $(KIND_NAMESPACE) --kubeconfig=$(MANAGED_CONFIG)
+	kubectl get all -n managed --kubeconfig=$(MANAGED_CONFIG)
+	kubectl get Policy.policy.open-cluster-management.io --all-namespaces --kubeconfig=$(MANAGED_CONFIG)
+	kubectl describe pods -n $(KIND_NAMESPACE) --kubeconfig=$(MANAGED_CONFIG)
+	kubectl logs $$(kubectl get pods -n $(KIND_NAMESPACE) -o name --kubeconfig=$(MANAGED_CONFIG) | grep $(IMG)) -n $(KIND_NAMESPACE) --kubeconfig=$(MANAGED_CONFIG)
```

Comment:
This won't work now that we are running the controller outside of the cluster.

Reply:
Shoot--you're right. I'll see about logging to a file and dumping the file.